tests/test-cases/Group1-Docker-Commands/1-02-Docker-Pull.robot

```diff
@@ -16,9 +16,85 @@
+    [Return]  ${output}
+
+Get Container Address
+    [Arguments]  ${container-name}  ${docker-ps}
+    ${rc}  ${container}=  Run And Return Rc And Output  echo '${docker-ps}' | grep ${container-name} | awk '{print $(NF-1)}' | cut -d- -f1
+    Should Be Equal As Integers  ${rc}  0
+    Log  ${container}
+    [Return]  ${container}
+
+Pull And MITM Prepared Image
+    [Arguments]  ${vch2-params}  ${registry}
+    ${rc}  ${output}=  Run And Return Rc And Output  docker ${vch2-params} pull ${registry}/busybox
+    Log  ${output}
+
+Enable SSH On MITMed VCH
+    ${rc}  ${thumbprint}=  Run And Return Rc And Output  govc about.cert -k -json | jq -r .ThumbprintSHA1
+    Log  ${thumbprint}
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}  ${output}=  Run And Return Rc And Output  bin/vic-machine-linux debug --rootpw password --target %{TEST_URL}%{TEST_DATACENTER} --password %{TEST_PASSWORD} --name VCH-XPLT --user %{TEST_USERNAME} --compute-resource %{TEST_RESOURCE} --enable-ssh --thumbprint=${thumbprint}
+    Should Be Equal As Integers  ${rc}  0
+    Log  ${output}
+
+Check For Injected Binary
+    [Arguments]  ${vch2-IP}
+    ${rc}  ${output}=  Run And Return Rc And Output  sshpass -ppassword ssh ${vch2-IP} -lroot -C -oStrictHostKeyChecking=no "ls /tmp | grep pingme"
+    Log  ${output}
+    Should Not Contain  ${output}  pingme
+
+Deploy Proxified VCH
+    [Arguments]  ${registry}  ${mitm}
+    ${br1}=  Get Environment Variable  BRIDGE_NETWORK
+    Create Unique Bridge Network
+    # BRIDGE_NETWORK gets overwritten by Create Unique Bridge Network. Assign original value to BRIDGE_NETWORK_2 for removal during suite teardown
+    Set Environment Variable  BRIDGE_NETWORK_2  ${br1}
+
+    # Run VIC Machine Command assumes %{VCH-NAME}
+    # Install VIC Appliance on Test Server assumes a bunch of environment variables
+    # We're just going to eschew helpers and install this VCH manually to avoid mutating hidden environmental state which is difficult to debug
+    ${rc}  ${output}=  Run And Return Rc And Output  bin/vic-machine-linux create --name=VCH-XPLT --target=%{TEST_URL}%{TEST_DATACENTER} --user=%{TEST_USERNAME} --password=%{TEST_PASSWORD} --force=true --compute-resource=%{TEST_RESOURCE} --timeout %{TEST_TIMEOUT} --bridge-network=%{BRIDGE_NETWORK} --container-network=%{PUBLIC_NETWORK}:public --public-network=%{PUBLIC_NETWORK} %{VICMACHINETLS} --image-store=%{TEST_DATASTORE} --insecure-registry=http://${registry} --http-proxy http://${mitm}
+    Log  ${output}
+
+    Should Be Equal As Integers  ${rc}  0
+    ${rc}  ${vch2-params}=  Run And Return Rc And Output  echo '${output}' | grep -A1 "Connect to docker" | tail -n1 | cut -d' ' -f4- | sed 's/ info" $//g'
+
+    # this comment fixes syntax highlighting "
+
+    Log  ${output}
+    Should Be Equal As Integers  ${rc}  0
+
+    ${rc}  ${vch2-IP}=  Run And Return Rc And Output  echo '${output}' | grep -A1 "Published ports" | tail -n1 | awk '{print $NF}' | cut -d= -f2
+    Log  ${output}
```

Comment:
duplicate, not understanding if you are intentionally doing this for some reason?  The var hasn't changed since the first call?

Reply:
I had a lot of Log statements because this was a tough test to get working. Then I did a big refactor so that it was actually readable, because it became clear to me that the whole test was pretty opaque. A side effect of this, and that I didn't remove the Log statements because generally I think they're valuable, is that we got a couple dupes and a couple that aren't in places that make sense. Thanks for finding them in review.